datadog_checks_dev/datadog_checks/dev/docker.py

```diff
@@ -22,4 +32,219 @@ def get_container_ip(container_id_or_name):
         container_id_or_name
     ]
 
-    return run_command(command, capture='out').stdout.strip()
+    return run_command(command, capture='out', check=True).stdout.strip()
+
+
+def compose_file_active(compose_file):
+    command = ['docker-compose', '-f', compose_file, 'ps']
+    lines = run_command(command, capture='out', check=True).stdout.splitlines()
+
+    for i, line in enumerate(lines, 1):
+        if set(line.strip()) == {'-'}:
+            return len(lines[i:]) >= 1
+
+    return False
+
+
+@contextmanager
+def docker_run(
+    compose_file=None,
+    up=None,
+    down=None,
+    endpoints=None,
+    log_patterns=None,
+    conditions=None,
+    env_vars=None,
+    wrapper=None
+):
+    if compose_file is not None and up is not None:
```

Comment:
you can reduce if statement like this `if compose_file and up:`. It applies in other places here